src/gf.c

```diff
@@ -1485,17 +1485,27 @@ static void update_max_args(jl_methtable_t *mt, jl_value_t *type)
         mt->max_args = na;
 }
 
-JL_DLLEXPORT int jl_debug_method_invalidation = 0;
+jl_array_t *_jl_debug_method_invalidation = NULL;
+JL_DLLEXPORT jl_value_t *jl_debug_method_invalidation(int state)
+{
+    if (state) {
+        if (_jl_debug_method_invalidation)
+            return (jl_value_t*) _jl_debug_method_invalidation;
+        JL_GC_PUSH1(&_jl_debug_method_invalidation);
+        _jl_debug_method_invalidation = jl_alloc_array_1d(jl_array_any_type, 0);
+        JL_GC_POP();
```

Comment:
```suggestion
        _jl_debug_method_invalidation = jl_alloc_array_1d(jl_array_any_type, 0);
```
(this frame is illogical, since there's no use of the value protected inside it)